### PR TITLE
Add section on events dispatched by SSE extension to SSE documentation

### PR DESCRIPTION
**Pull request by vlence**
Describe the events dispatched by the SSE event in the style of how the HTMX events are documented.

### CODE REVIEW

File: src/sse/README.md [hunk 1]
```diff
@@ -113,6 +113,56 @@ Previous versions of htmx used a built-in tag `hx-sse` to implement Server Sent
...
+* `detail.elt` - The element on which the SSE connection was setup. This is the element which has the `sse-connect` attribute.
+* `detail.source` - The [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) object.
+
+#### `htmx:sseError`
+
+This event is dispatched when an SSE connection could not be established.
+
+##### Details
+
+* `detail.elt` - The element on which the SSE connection was setup. This is the element which has the `sse-connect` attribute.
```
Comment (Telroshan) [inline src/sse/README.md:143]: This doesn't seem to be the case here;
https://github.com/bigskysoftware/htmx-extensions/blob/8b03c9136e182954ccfc5a35b0a73cf7f59b6f17/src/sse/sse.js#L196

The event is fired with triggerErrorEvent, which [doesn't seem to include elt](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L2210) as opposite to `triggerEvent`

File: src/sse/README.md [hunk 2]
```diff
@@ -113,6 +113,56 @@ Previous versions of htmx used a built-in tag `hx-sse` to implement Server Sent
...
+* `detail.source` - The [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) object.
+
+#### `htmx:sseError`
+
+This event is dispatched when an SSE connection could not be established.
+
+##### Details
```
Comment (Telroshan) [inline src/sse/README.md:141]: You also might want to add a reference to `error` here
https://github.com/bigskysoftware/htmx-extensions/blob/8b03c9136e182954ccfc5a35b0a73cf7f59b6f17/src/sse/sse.js#L196

File: src/sse/README.md [hunk 3]
```diff
@@ -113,6 +113,56 @@ Previous versions of htmx used a built-in tag `hx-sse` to implement Server Sent
...
+
+* `detail.elt` - The element on which the SSE connection was setup. This is the element which has the `sse-connect` attribute.
+* `detail.source` - The [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) object.
+
+#### `htmx:sseBeforeMessage`
+
+This event is dispatched just before the SSE event data is swapped into the DOM.
```
Comment (Telroshan) [inline src/sse/README.md:148]: Might be worth mentioning here that calling `event.preventDefault` will abort the SSE swap, as I imagine this event would be primarily used for this purpose

File: src/sse/README.md [hunk 4]
```diff
@@ -113,6 +113,56 @@ Previous versions of htmx used a built-in tag `hx-sse` to implement Server Sent
...
+
+##### Details
+
+* `detail.elt` - The swap target.
+* `detail.data` - The data sent in the SSE message.
+* `detail.target` - The [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) object.
```
Comment (Telroshan) [inline src/sse/README.md:154]: https://github.com/bigskysoftware/htmx-extensions/blob/8b03c9136e182954ccfc5a35b0a73cf7f59b6f17/src/sse/sse.js#L114

- Might be worth mentioning here that `detail` is the native [`EventSource message event`](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/message_event#event_properties), on top of which htmx adds the `elt` property. Thus `data` is from the event source's event itself
- `target` on the other hand probably doesn't exist unless I missed something? I don't see the extension specifying it, nor do I see the MDN doc mentioning such a property (`source` on the other hand, does exist)

File: src/sse/README.md [hunk 5]
```diff
@@ -113,6 +113,56 @@ Previous versions of htmx used a built-in tag `hx-sse` to implement Server Sent
...
+
+#### `htmx:sseMessage`
+
+This event is dispatched after the SSE event data has been swapped into the DOM.
+
+##### Details
+
+* `detail.elt` - The swap target.
+* `detail.data` - The data sent in the SSE message.
+* `detail.target` - The [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) object.
```
Comment (Telroshan) [inline src/sse/README.md:164]: Same remarks as `sse:beforeMessage` as it's the same `detail` context
https://github.com/bigskysoftware/htmx-extensions/blob/8b03c9136e182954ccfc5a35b0a73cf7f59b6f17/src/sse/sse.js#L118

**Review by vlence — COMMENTED** [review]
Made fixes as recommended by @Telroshan

**Comment by Telroshan** [comment]
Nice, thank you!

**Pull request merged**